classes/SnippetManager.php

```diff
@@ -35,13 +35,58 @@ public function listSnippets($theme)
         $themeSnippets = $this->listThemeSnippets($theme);
         $componentSnippets = $this->listComponentSnippets();
 
-        $snippets = array_merge($themeSnippets, $componentSnippets);
+        $this->snippets = array_merge($themeSnippets, $componentSnippets);
+
+        /*
+         * @event pages.snippet.listSnippets
+         * Gives the ability to manage snippet list dynamically.
+         *
+         * Example usage to add a snippet to the list:
+         *
+         * Event::listen('pages.snippet.listSnippets', function(&$manager) {
```

Comment:
@tomaszstrojny `$manager` does not need the pass-by-reference (`&`) character, as it is a class instance and is already passed by reference.
```suggestion
         * Event::listen('pages.snippets.listSnippets', function ($manager) {
```